Framework/Geometry/test/ComponentHelperTest.h

```diff
@@ -107,6 +110,59 @@ class ComponentHelperTest : public CxxTest::TestSuite {
     TS_ASSERT_DELTA(newRot.imagK(), expectedRot.imagK(), 1e-12);
   }
 
+  void test_absolute_planar_rotation_takes_into_account_grandparent() {
+    const int banks = 1;
+    const int pixels = 1;
+    auto base =
+        ComponentCreationHelper::createTestInstrumentRectangular(banks, pixels);
+    auto pmap = boost::make_shared<ParameterMap>();
+    Instrument instrument(base, pmap);
+
+    const auto bank = instrument.getComponentByName("bank1");
+    const auto pixel = instrument.getDetector(1);
+
+    Quat rootRot(100.0, V3D(1, 2, 3));
```

Comment:
`Quat` has quite a few constructors. A minor improvement might be to comment or provide named variables so that it's clear the construction arguments are in this case rotation angle and axis.